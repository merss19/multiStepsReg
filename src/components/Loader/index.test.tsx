import {shallow} from 'enzyme';
import {Loader} from './index';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
let text;
let isError;
const start = () => {
  beforeEach(() => {
    text = 'Подождите',
      isError = false
  });
  const props = {
    text: text,
    isError: isError
  };
  const buildComponent = () => {
    return shallow(<Loader {...props}/>)
  };
  return {
    props,
    buildComponent
  }
};
describe('Component Loader', () => {
  it('renders component', () => {
    const component = start().buildComponent();
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(start().buildComponent());
    expect(tree).toMatchSnapshot();
  });
  it('render loader', () => {
    const component = start().buildComponent();
    expect(component.find('.loader')).toHaveLength(1);
    expect(component.find('.loading__desc')).toHaveLength(0);
  });
  it('render text', () => {
    isError = true;
    const component = start().buildComponent();
    expect(component.find('h2').text()).toEqual(start().props.text);
    expect(component.find('.loading__desc')).toHaveLength(1);
    expect(component.find('.loader')).toHaveLength(0);
    text = undefined;
    component.setProps({text: text});
    expect(component.find('h2').text()).toEqual('Загружается');
  });
});
