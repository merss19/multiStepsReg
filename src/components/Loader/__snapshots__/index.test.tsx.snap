// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Loader renders correctly 1`] = `
<div
  className="loading text-center"
>
  <div
    className="loader loader--main"
  />
</div>
`;
